staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go

```diff
@@ -122,8 +123,9 @@ func NewCommandStartWardleServer(ctx context.Context, defaults *WardleServerOpti
 		},
 	}))
 	utilruntime.Must(utilversion.DefaultComponentGlobalsRegistry.Register(apiserver.WardleComponentName, wardleEffectiveVersion, wardleFeatureGate))
-	_, _ = utilversion.DefaultComponentGlobalsRegistry.ComponentGlobalsOrRegister(
-		utilversion.DefaultKubeComponent, utilversion.DefaultKubeEffectiveVersion(), utilfeature.DefaultMutableFeatureGate)
+	kubeVersion := version.MustParse(baseversion.DefaultKubeBinaryVersion).AddMinor(1)
```

Comment:
why AddMinor?

Reply:
Wardle v1.1 maps to default kube version (v1.31 for now):
https://github.com/kubernetes/kubernetes/blob/665479ce4964b0c8232b4f9c630dc28c9d6f12dd/staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go#L68

So wardle v1.2 (declared [here](https://github.com/kubernetes/kubernetes/blob/665479ce4964b0c8232b4f9c630dc28c9d6f12dd/staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go#L116)) will map to default kube version + 1 minor.

Reply:
not sure I can follow. This needs a go doc saying exactly what is done here and why.

Reply:
Also the lines above look like magic. Please document.

Reply:
@sttts Added some documentation. PTAL.

cc. @siyuanfoundation @jpbetz

Reply:
Thanks. A lot better! 👍

Reply:
But even with the docs, I don't understand why we are defining a relationship between kube and wardle. Sample-apiserver is not a test fixture. It's an example how to write a proper aggregated apiserver that integrates well. Why would some 3rdparty relate kube and wardle version in the way it is done here?

Reply:
My naive assumption: we are building against Kube 1.31 libraries here (as soon it is released). So our effective 1.2 wardle version should match 1.31. So why plus one?

Reply:
We want to test the case the kube emulation version is set to `N` and `N-1`. Right now kube emulation version  can only be set to `1.31` if binary version is `1.31` because it is introduced in `1.31` and cannot go back to `1.30`. So we have to tie the wardle binary version to kube binary version `1.32` for testing `N` and `N-1`

Reply:
>> Sample-apiserver is not a test fixture.
> We want to test the case the kube emulation version is set to N and N-1

Which means we are using this as a test fixture?

Reply:
moved the +1 logic into integration tests so it is less confusing in the regular code.